### PR TITLE
fix(react/pagination): render page number twice when `pagesUnknown`

### LINKED ISSUE #12252: [Bug]: Pagination shows redundant information when `pagesUnknown` is set to true.
### Package

@carbon/react, carbon-components-react

### Browser

Chrome, Firefox, Edge

### Package version

v1.14

### Description

In pagination component, when `pagesUnknown` prop has been set to `true`, the page number is shown twice. 

e.g.
1 page 1
If I change the page number via `select`, the page number also updates as such:
2 page 2
3 page 3
...etc,

The first number is the page option selected from the select element. Then the same information is repeated in the following text.
Shouldn't it just be `Page <select element with number>` [TEXT (SELECT ELEMENT)]? Is there a reason for repeating the page number?  
  
I'm not sure if this counts as a design defect, but this issue was raised in [Carbon Angular](https://github.com/IBM/carbon-components-angular/issues/2199), just forwarding it here so we know what steps to take.

### Screenshots

image
image
image

### Steps to reproduce

Go to the following storybook (props have been already set, so no changes required):

https://react.carbondesignsystem.com/?path=/story/components-pagination--default&args=pagesUnknown:true

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by 61130061**
Closes #12252 

According to #12252, when pagesUnknown is `true`, pagination render page number as same as selector which does not make sense to render this information twice as shown in figure before. So, I change page number (pageText) to page label in front of the selector which makes more sense and make pagination looks cleaner. Also, change `pageText` prop from `function` to `string` so user can change the page label.

before:
before_figure

after:
after_figure

#### Changelog

**New**

- {{new thing}}

**Changed**

- hide/show page label and total page number depend on pagesUnknown
- `pageText` default change to `page`
- typeof pageText change from `function` to `string`
- test case of pagination pagesUnknown = `true` change to expect just `page` since switch from page number to page label as shown in figures
- change border-left style form `selector` to `pagination__right`

**Removed**

- {{removed thing}}

#### Testing / Reviewing

- All test passed!

### CODE REVIEW

**Comment by 61130061** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by tw15egan** [comment]
@61130061 We'll need to keep this a function so that teams can pass down a function for translation keys. I think this can be fixed just by passing down ` pageText={() => ''}` to the `Pagination` component as-is. It seems like there is a slight styling issue in this scenario, but can be fixed by just adding the following CSS to `_pagination.scss`

```scss
.#{$prefix}--pagination__right .#{$prefix}--pagination__text:empty {
    margin: 0;
}

File: packages/react/src/components/Pagination/next/Pagination.js [hunk 1]
```diff
@@ -67,7 +67,9 @@ const Pagination = React.forwardRef(function Pagination(
     pageSize: controlledPageSize,
     pageSizeInputDisabled,
     pageSizes: controlledPageSizes,
-    pageText = (page) => `page ${page}`,
+    pageText = (page, pagesUnknown) => {
+      return `page ${pagesUnknown ? '' : page}`;
+    },
```
Comment (tay1orjones) [inline packages/react/src/components/Pagination/next/Pagination.js]: Let's match the style of the rest of the translatable function strings in this component and use an implicit return

```suggestion
    pageText = (page, pagesUnknown) => `page ${pagesUnknown ? '' : page}`,
```

**Review by tay1orjones — APPROVED** [review]
One small suggested change, but overall this looks good to me. Thanks for contributing this fix!

@61130061 There are conflicts that need resolved before this can be merged

**Comment by 61130061** [comment]
@tay1orjones Thank you for your approval and suggestion! I commit another `Pagination.js` with the same problem.  I think all conflicts are resolved now.

**Comment by tay1orjones** [comment]
@61130061 There's still some merge conflicts here

**Comment by tay1orjones** [comment]
I just fixed the merge conflicts.

@tw15egan the merge was a bit hairy, but I think I got it all. Could you review again once the deploy preview is updated?

File: packages/react/src/components/Pagination/next/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,421 @@
+/**
```
Comment (tw15egan) [inline packages/react/src/components/Pagination/next/Pagination.js]: I believe this file was removed in https://github.com/carbon-design-system/carbon/pull/12115

File: packages/react/src/components/Pagination/next/__tests__/Pagination-test.js [hunk 3]
```diff
@@ -0,0 +1,526 @@
+/**
```
Comment (tw15egan) [inline packages/react/src/components/Pagination/next/__tests__/Pagination-test.js]: I believe this file was removed in https://github.com/carbon-design-system/carbon/pull/12115

**Review by tw15egan — CHANGES_REQUESTED** [review]
I think most of the merge was correct, I think these two just got added back in inadvertently

**Comment by tay1orjones** [comment]
@tw15egan Thanks for the catch - I just pushed a commit to remove those files

**Review by tw15egan — APPROVED** [review]

**Pull request merged**